tests/unit/test_dataframe.py

```diff
@@ -200,3 +200,20 @@ def test_same_joins_should_generate_same_queries(join_type):
     )
 
     assert df1.join(df2, how=join_type).queries == df1.join(df2, how=join_type).queries
+
+
+def test_statement_params():
+    mock_connection = mock.create_autospec(ServerConnection)
+    mock_connection._conn = mock.MagicMock()
+    session = snowflake.snowpark.session.Session(mock_connection)
+    session._conn._telemetry_client = mock.MagicMock()
+    df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
+    statement_params = {"param1": "val", "param2": 0, "param3": True}
+    df._statement_params = statement_params
```

Comment:
should we add a better way to set `_statement_params`. Like adding it in `__init__` or adding a method `dataframe.set_statement_params(params)`

Reply:
@sfc-gh-yixie @sfc-gh-sfan, what do you think?

Reply:
I see. thanks for clarification